clients/python-wrapper/pylotl/object_io.py

```diff
@@ -0,0 +1,114 @@
+from typing import Optional, Literal, Union, Iterable, AsyncIterable
+
+from lakefs_sdk.exceptions import NotFoundException
+from pylotl.client import Client
+from pylotl.exceptions import ObjectExistsException, UnsupportedOperationException, ObjectNotFoundException
+
+_RANGE_STR_TMPL = "bytes={start}-{end}"
+
+# Type to support both strings and bytes in addition to streams (reference: httpx._types.RequestContent)
+UploadContentType = Union[str, bytes, Iterable[bytes], AsyncIterable[bytes]]
+
+
+class ReadableObject:
+    """
+    ReadableObject provides read-only functionality for lakeFS objects with IO semantics.
+    This Object is instantiated and returned on open() methods for immutable reference types (Commit, Tag...)
+    """
+    _client: Client
+    _repo: str
+    _ref: str
+    _path: str
+    _pos: int
+    _pre_sign: bool
+
+    def __init__(self, client: Client, repository: str, reference: str, path: str, pre_sign: bool = None) -> None:
```

Comment:
```suggestion
    def __init__(self, repository: str, reference: str, path: str, pre_sign: Optional[bool] = None, client: Optional[Client] = None) -> None:
```

Please use this style everywhere - it would allow anyone looking to directly init this class to do so easily with the default client